### ISSUE TITLE
Fast preconditioner caches its build payload on the Module instance and skips invalidation when parameters change

### ISSUE THREAD

**Issue by sachncs** [sachncs/xaker#37]
## Problem

`xaker/solver/precond.py:133-193` defines `class Fast(nn.Module)`, whose `build` method caches the most recent preconditioner payload on the module:

```python
def build(self, kernel, lam, length):
    if (
        self.cache is not None
        and self.freq > 0
        and int(self.iter[0]) % self.freq != 0
    ):
        return self.cache
    ...
    self.cache = Cache(data=payload)
    return self.cache
```

This caches the *frozen tensor values* computed from `lr_base[:, :length, :] * softplus(lr_imp)` — but `lr_base` and `lr_imp` are `nn.Parameter` instances that change on every optimizer step. The cache compares only `int(self.iter[0]) % self.freq`. There is **no** invalidation when the parameter values change. Consequently:

- During training, if `freq > 1` (e.g. `Config.freq=4`), the cached `diag` and `lr` tensors used by `apply_pre` are stale by up to `freq` training steps. They were computed from the parameter values at the last build, but `lr_base` and `lr_imp` have since been updated by SGD.
- Worse, because the cache is a plain Python attribute on an `nn.Module`, it is **not** part of `state_dict`, **not** tracked by autograd, and **not** tracked by `optimizer.zero_grad()`. The cache therefore masks parameter updates from the preconditioner contract.

Tests in `tests/test_fused.py:99-106` (TestFused::test_grad) do not exercise the cached path because they only do a single forward pass (so `self.iter[0] == 0` and the cache branch is never taken).

## Evidence

- `xaker/solver/precond.py:133-193` — `Fast.__init__`, `Fast.build`, `Fast.apply_pre`. The cache and the `iter` step-counter are written **only** here.
- `xaker/solver/precond.py:162-164` — `register_buffer(\"iter\", torch.zeros(1, dtype=torch.long))` *plus* a duplicate type annotation `self.iter: torch.Tensor` — both forms present.
- `xaker/attention/fused.py:170` — `fast.iter.add_(1)` after each forward; this is the only bump site.
- `xaker/config.py:55` — `freq: int = 1`. Default `freq=1` *does* rebuild every step, so the cache branch is dormant under default settings. The bug only manifests when a user sets `freq > 1`.

```python
>>> # Demonstration:
>>> cfg = Config(dim=64, heads=4, precond=\"fast\", rank=4, freq=2)
>>> fast = Make(cfg)
>>> k = torch.softmax(torch.randn(2, 4, 32, 32), dim=-1)
>>> lam = torch.tensor(0.1)
>>> a = fast.build(k, lam, 32)
>>> # Modify lr_base without invalidating cache:
>>> with torch.no_grad():
...     fast.lr_base.fill_(0.0)
>>> b = fast.build(k, lam, 32)
>>> a_data, b_data = a.data, b.data
>>> torch.equal(a_data[1], b_data[1])   # True — cache hit used stale data
```

## Root Cause

Caching the preconditioner payload as a Python attribute (instead of recomputing per step or hashing the inputs) trades freshness for speed. The freshness signal (a step counter compared with `freq`) is not equivalent to "params changed".

## Impact

- Silent correctness issue when `freq > 1`. Convergence trajectories may diverge from what `Fused.attend`'s math docstring describes.
- `fast.iter.add_(1)` increments *globally on the preconditioner* rather than per (batch, head). Two Fused blocks in a Transformer share the same step-counter; they shouldn't, because they each have their own preconditioner instance.
- The cache attribute is invisible to `state_dict`, so checkpoint/restore cannot preserve the cached payload — actually this is the *intended* behavior but means a resumed training run silently uses a stale cache after the first build.

## Recommendation

1. Invalidate the cache based on `lr_base.data_ptr()` (or, more robustly, hash a few signature statistics) plus the step counter; on mismatch, rebuild.
2. Replace the dual `register_buffer(\"iter\", ...)` + `self.iter: torch.Tensor` pattern with a single annotation.
3. Document in `Config.freq` and `Config.precond` docstring that `freq > 1` is currently a no-op for the Fast preconditioner (it does enable some savings but should be considered beta).
4. Add a regression test in `tests/test_solver.py` that constructs `Fast`, calls `build` twice with mutated `lr_base`, and asserts the second `Cache.data` differs.

## Acceptance Criteria

- [ ] Mutation of `lr_base` between two `Fast.build` calls invalidates the cache when `freq > 1`.
- [ ] Single Python attribute describes the cache (no separate `register_buffer` step counter).
- [ ] `Config.freq > 1` either rebuilds correctly or is documented as experimental.
- [ ] Test in `tests/test_solver.py` asserts cache invalidation.

## Verification

```bash
python -c \"
import torch
from xaker.config import Config
from xaker.solver.precond import Make
cfg = Config(dim=64, heads=4, precond='fast', rank=4, freq=2)
fast = Make(cfg)
k = torch.softmax(torch.randn(2, 4, 32, 32), dim=-1)
a = fast.build(k, torch.tensor(0.1), 32)
with torch.no_grad():
    fast.lr_base.fill_(0.0)
b = fast.build(k, torch.tensor(0.1), 32)
print(torch.equal(a.data[1], b.data[1]))  # expected: False after fix
\"
```

## Dependencies

None.

## Risk

Low-medium. Behaviour change for users with `freq > 1`; default `freq=1` is unaffected.
